src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -86,24 +104,85 @@ def add_helm_repo(kube_config, kube_context):
+    uri_parameters = ["releaseTrain={}".format(release_train)]
     try:
-        response = requests.post(get_chart_location_url, params=query_parameters, headers=header_parameters)
+        r = send_raw_request(cmd.cli_ctx, 'post', get_chart_location_url, uri_parameters=uri_parameters, resource='https://management.core.windows.net/')
     except Exception as e:
         telemetry.set_exception(exception=e, fault_type=consts.Get_HelmRegistery_Path_Fault_Type,
                                 summary='Error while fetching helm chart registry path')
-        raise CLIError("Error while fetching helm chart registry path: " + str(e))
-    if response.status_code == 200:
-        return response.json().get('repositoryPath')
-    telemetry.set_exception(exception=str(response.json()), fault_type=consts.Get_HelmRegistery_Path_Fault_Type,
-                            summary='Error while fetching helm chart registry path')
-    raise CLIError("Error while fetching helm chart registry path: {}".format(str(response.json())))
+        raise CLIError("Error while fetching helm chart registry path: " + str(e) + "\nPlease file an issue on https://github.com/Azure/azure-arc-kubernetes-preview/issues")
+    if r.content:
+        try:
+            return r.json().get('repositoryPath')
+        except Exception as e:
+            telemetry.set_exception(exception=e, fault_type=consts.Get_HelmRegistery_Path_Fault_Type,
+                                    summary='Error while fetching helm chart registry path')
+            raise CLIError("Error while fetching helm chart registry path from JSON response: " + str(e) + "\nPlease file an issue on https://github.com/Azure/azure-arc-kubernetes-preview/issues")
+    else:
+        telemetry.set_exception(exception='No content in response', fault_type=consts.Get_HelmRegistery_Path_Fault_Type,
+                                summary='No content in acr path response')
+        raise CLIError("No content was found in helm registry path response." + "\nPlease file an issue on https://github.com/Azure/azure-arc-kubernetes-preview/issues")
+
+
+def arm_exception_handler(ex, fault_type, summary, return_if_not_found=False):
+    if isinstance(ex, AuthenticationError):
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception=ex, fault_type=fault_type, summary=summary)
+        raise CLIError("Authentication Error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
+
+    if isinstance(ex, TokenExpiredError):
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception=ex, fault_type=fault_type, summary=summary)
+        raise CLIError("Token Expired Error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
+
+    if isinstance(ex, HttpOperationError):
+        status_code = ex.response.status_code
+        if status_code == 404 and return_if_not_found:
+            return
+        if status_code // 100 == 4:
```

Comment:
>status_code // 100 == 4 [](start = 11, length = 23)

What does this mean?

Reply:
'//' is the floor division operator in python.